examples/chip-tool/commands/clusters/ModelCommand.cpp

```diff
@@ -37,6 +37,25 @@ void DispatchSingleClusterCommand(chip::ClusterId aClusterId, chip::CommandId aC
         "Default DispatchSingleClusterCommand is called, this should be replaced by actual dispatched for cluster commands");
 }
 
+CHIP_ERROR WaitForSessionSetup(chip::Controller::Device * device)
+{
+    constexpr time_t kWaitPerIteration = 1;
+    constexpr uint16_t kIterationCount = 5;
+
+    struct timespec sleep_time;
+    sleep_time.tv_sec  = kWaitPerIteration;
+    sleep_time.tv_nsec = 0;
+
+    for (uint32_t i = 0; i < kIterationCount && device->IsSessionSetupInProgress(); i++)
+    {
+        nanosleep(&sleep_time, nullptr);
```

Comment:
could we instead have some sort of notification from the device of 'session done'?

May not care much for test tools, however I imagine most other commisioners want to avoid sleeps.

Reply:
Added a TODO for this. This is big enough change to handle in a separate PR.